From a79e88dd2ca6686dca77c9c53c3e12c031347348 Mon Sep 17 00:00:00 2001
Message-Id: <a79e88dd2ca6686dca77c9c53c3e12c031347348.1649663517.git.geliang.tang@suse.com>
From: Martin KaFai Lau <kafai@fb.com>
Date: Thu, 11 Feb 2021 18:10:37 -0800
Subject: [PATCH] bpf: selftests: Add non function pointer test to struct_ops
Git-commit: a79e88dd2ca6686dca77c9c53c3e12c031347348
Patch-mainline: v5.12-rc2
References: bsn#127
Modified-by-SEL: No


This patch adds a "void *owner" member.  The existing
bpf_tcp_ca test will ensure the bpf_cubic.o and bpf_dctcp.o
can be loaded.

Signed-off-by: Martin KaFai Lau <kafai@fb.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Acked-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20210212021037.267278-1-kafai@fb.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/testing/selftests/bpf/bpf_tcp_helpers.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/testing/selftests/bpf/bpf_tcp_helpers.h b/tools/testing/selftests/bpf/bpf_tcp_helpers.h
index 6a9053162cf2..91f0fac632f4 100644
--- a/tools/testing/selftests/bpf/bpf_tcp_helpers.h
+++ b/tools/testing/selftests/bpf/bpf_tcp_helpers.h
@@ -177,6 +177,7 @@ struct tcp_congestion_ops {
 	 * after all the ca_state processing. (optional)
 	 */
 	void (*cong_control)(struct sock *sk, const struct rate_sample *rs);
+	void *owner;
 };
 
 #define min(a, b) ((a) < (b) ? (a) : (b))
-- 
2.34.1

